apps/todo-app/src/app/interceptors/api-key.interceptor.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { environment } from '../../environments/environment';
+
+@Injectable()
+export class ApiKeyInterceptor implements HttpInterceptor {
+  intercept(
+    request: HttpRequest<unknown>,
+    next: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    const endpoint = request.url.split('/api/');
```

Comment:
why you split the endpoint?

Reply:
this was for adding api key into url, I refactored it to be more readable